statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        n = data.shape[0]
+        p = data.shape[1]
+        self.data = np.asarray(data).reshape(n,p)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, sampling_method):
+        """
+        converts cluster, strata, etc to an numpy array
+        gets rid of column from data if possible
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        sampling_method : one of cluster, strata, or prob_weight
+
+        Returns
+        -------
+        A nx1 numpy array
+        """
+
+        # check if str, num, or list
+        if isinstance(sampling_method, str):
```

Comment:
Maybe some of the init data handling can be delegated to the standard framework by subclassing base.Model.
Or/and we could use a survey data class with the survey structure similar to R and Stata.

But those API/data structure decisions could be postponed until a few of the statistics and models are implemented.